### PR TITLE
WRF-Hydro

### PR DESCRIPTION
**Pull request by afraz-ahmed**
For INC1162729 - `WRF-Hydro-5.1.1-foss-2020b.eb WRF-Hydro_gfortran10_netcdff.patch`

* [x] Assigned to reviewers (usually everyone in apps team)

Default:
* [x] EL8-cascadelake
* [x] EL8-haswell

### CODE REVIEW

File: easybuild/easyconfigs/w/WRF-Hydro/WRF-Hydro-5.1.1-foss-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+skipsteps = ['configure', 'build']
+
+preinstallopts = 'cp template/setEnvar.sh . && '
+preinstallopts += 'sed -i s/SPATIAL_SOIL=0/SPATIAL_SOIL=1/ setEnvar.sh && '
+preinstallopts += 'echo "export WRFIO_NCD_LARGE_FILE_SUPPORT=1" >> setEnvar.sh && '
+preinstallopts += 'export NETCDF_INC=$EBROOTNETCDF/include && '
+preinstallopts += 'export NETCDF_LIB=$EBROOTNETCDF/lib && '
+preinstallopts += 'export NETCDFF_INC=$EBROOTNETCDFMINFORTRAN/include &&'
+preinstallopts += 'export NETCDFF_LIB=$EBROOTNETCDFMINFORTRAN/lib && '
+preinstallopts += './configure 2 && '  # option 2 for gfortran
```
Comment (branfosj) [inline easybuild/easyconfigs/w/WRF-Hydro/WRF-Hydro-5.1.1-foss-2020b.eb:43]:
```suggestion
skipsteps = ['build']

_netcdfvars = "NETCDF_INC=$EBROOTNETCDF/include NETCDF_LIB=$EBROOTNETCDF/lib "
_netcdfvars += "NETCDFF_INC=$EBROOTNETCDFMINFORTRAN/include NETCDFF_LIB=$EBROOTNETCDFMINFORTRAN/lib "

preconfigopts = 'cp template/setEnvar.sh . && '
preconfigopts += 'sed -i s/SPATIAL_SOIL=0/SPATIAL_SOIL=1/ setEnvar.sh && '
preconfigopts += 'echo "export WRFIO_NCD_LARGE_FILE_SUPPORT=1" >> setEnvar.sh && '
preconfigopts += _netcdfvars
configopts = '2'

preinstallopts = _netcdfvars
```

Can you try with that and see if it works?

**Review by branfosj — CHANGES_REQUESTED** [review]
If my suggested change works then it is a bit cleaner and I'd prefer it.

File: easybuild/easyconfigs/w/WRF-Hydro/WRF-Hydro-5.1.1-foss-2020b.eb (hunk 1 above)
Comment (gyear) [reply easybuild/easyconfigs/w/WRF-Hydro/WRF-Hydro-5.1.1-foss-2020b.eb:43]: Get this Error
./configure: line 72: [[: --prefix=/rds/bear-apps/devel/2020b/yearwoog-eb-20b/el8/el8-cas/software/wrf-hydro/5.1.1-foss-2020b: attempted assignment to non-variable (error token is "=/rds/bear-apps/devel/2020b/yearwoog-eb-20b/el8/el8-cas/software/wrf-hydro/5.1.1-foss-2020b")
./configure: line 78: [[: --prefix=/rds/bear-apps/devel/2020b/yearwoog-eb-20b/el8/el8-cas/software/wrf-hydro/5.1.1-foss-2020b: attempted assignment to non-variable (error token is "=/rds/bear-apps/devel/2020b/yearwoog-eb-20b/el8/el8-cas/software/wrf-hydro/5.1.1-foss-2020b")
./configure: line 84: [[: --prefix=/rds/bear-apps/devel/2020b/yearwoog-eb-20b/el8/el8-cas/software/wrf-hydro/5.1.1-foss-2020b: attempted assignment to non-variable (error token is "=/rds/bear-apps/devel/2020b/yearwoog-eb-20b/el8/el8-cas/software/wrf-hydro/5.1.1-foss-2020b")
./configure: line 97: [[: --prefix=/rds/bear-apps/devel/2020b/yearwoog-eb-20b/el8/el8-cas/software/wrf-hydro/5.1.1-foss-2020b: attempted assignment to non-variable (error token is "=/rds/bear-apps/devel/2020b/yearwoog-eb-20b/el8/el8-cas/software/wrf-hydro/5.1.1-foss-2020b")
No compiler selected. Exiting

Comment (branfosj) [reply easybuild/easyconfigs/w/WRF-Hydro/WRF-Hydro-5.1.1-foss-2020b.eb:43]: OK, revert the last change and we'll go with that.

**Review by branfosj — APPROVED** [review]

**Pull request merged**